static/js/tools/shared_util.ts

```diff
@@ -192,3 +196,30 @@ export function isChildPlaceOf(
 export function toTitleCase(str: string): string {
   return str.toLowerCase().replace(/\b(\w)/g, (s) => s.toUpperCase());
 }
+
+/**
+ * Given the dcid of a place, returns a promise with the rest of the place
+ * information
+ * @param placeDcid
+ */
+export function getNamedTypedPlace(
+  placeDcid: string
+): Promise<NamedTypedPlace> {
+  if (placeDcid === EARTH_NAMED_TYPED_PLACE.dcid) {
+    return Promise.resolve(EARTH_NAMED_TYPED_PLACE);
+  }
+  const placeTypePromise = axios
+    .get(`/api/place/type/${placeDcid}`)
```

Comment:
should we add a new call to flask that does both of these together? possibly with the results of parent places as well

Reply:
The new cache with all the parents, name, type information could be used here. Maybe add a TODO to use that

Reply:
We could, that might be more clean. But that would also mean we have to wait for one call to finish before we can call the other. Whereas by doing it this way, we get to call both asynchronously

Reply:
added TODO

Reply:
re: sequential calls: these calls between flask + mixer should be much faster than the async calls from client to flask, especially since flask and mixer are in the same container. and once it's computed once, it'll be cached for future calls.

but new cache will be much faster than that :)

Reply:
Ah, ok, that makes sense, I'll add the new flask endpoint then and we can use later use the new cache, but I'll do this in a separate PR.